### ISSUE TITLE
omitempty not working for empty time.Time values

### ISSUE THREAD

**Issue by beefsack** [rethinkdb/rethinkdb-go#176]
Hello @dancannon!

Just updated to the latest version and seeing something funny with empty `time.Time` values and `omitempty`, reproduction at: https://gist.github.com/beefsack/d6cbea75e0f7e31f4a70

The output of that script running locally:

``` bash
2015/04/16 18:22:53 Testing insert with empty Time
2015/04/16 18:22:53 Inserted with id 0ea091c5-64ba-4f62-8473-79803478e2a4
2015/04/16 18:22:53 Grabbing it back
2015/04/16 18:22:53 Got time 1754-08-30 22:43:41.128999711 +0000 +00:00
```

**Comment by dancannon** [comment]
Hi @beefsack,

Thanks for the great example code, it really helps when debugging. I have managed to track down the source of the issue (to the change to how times are encoded and how "empty values" are detected).

I should have a fix ready sometime tomorrow and this should hopefully be merged to master by around Wednesday next week.

**Comment by beefsack** [comment]
Awesome!  Thanks for looking into this one :+1:

**Comment by dancannon** [comment]
I have merged a fix to develop, as mentioned I will release this fix to master later this week with v0.7.1.

**Comment by beefsack** [comment]
Fantastic, have tested and working brilliantly, thanks for sorting that out!
